api-spec/transactions-api.yaml

```diff
@@ -7,6 +7,14 @@ paths:
   /transactions:
     post:
       operationId: newTransaction
+      parameters:
+        - in: header
+          name: X-Client-Id
+          schema:
+            type: string
+            pattern: 'IO|CHECKOUT|CHECKOUT_CART'
```

Comment:
Bikeshedding: enum?

Reply:
I changed the type to enumeration but the pattern is still needed since header is generated as string object and removing pattern descriptor will cause generated code to not have any kind of content validation